xcube/core/store/fs/store.py

```diff
@@ -321,7 +321,7 @@ def open_data(
         open_params_schema = self._get_open_data_params_schema(opener, data_id)
         assert_valid_params(open_params, name="open_params", schema=open_params_schema)
         fs_path = self._convert_data_id_into_fs_path(data_id)
-        return opener.open_data(fs_path, fs=self.fs, root=self.root, **open_params)
+        return opener.open_data(fs_path, fs=self.fs, **open_params)
```

Comment:
`fs_path = self._convert_data_id_into_fs_path(data_id)` already adds root to `fs_path`. 
`opener.open_data(fs_path, fs=self.fs, root=self.root, **open_params)` then adds root a second time to `fs_path` [here](https://github.com/xcube-dev/xcube/blob/main/xcube/core/store/fs/impl/dataset.py#L313).

To reproduce the error, run the following lines in a python console after directing into the xcube repo: 

```
from xcube.core.store import new_data_store


store = new_data_store("file", root="examples/serve/demo")
dataset = store.open_data(
    "sample-geotiff.tif",
    opener_id="dataset:geotiff:file"
)
```

@TejasMorbagal: you modified this line last. Can you remember if there was a good reason for it, which I oversee?

Reply:
looks like a bug! Surprised, this didn't get caught all this time. Please go ahead with your change.

Reply:
:)

Reply:
> Please go ahead with your change.

@TejasMorbagal please approve then.